python/chemiscope/input.py

```diff
@@ -245,46 +159,38 @@ def create_input(
-            # only include values for the subset of environments
-            # requested by the user
-            atoms_mask = [[False] * s["size"] for s in data["structures"]]
-            for environment in data["environments"]:
-                atoms_mask[environment["structure"]][environment["center"]] = True
-
-            atoms_mask = np.concatenate(atoms_mask)
-        else:
-            atoms_mask = None
-
-        for name, value in atom_properties(frames, composition, atoms_mask).items():
-            _validate_property(name, value)
-            for name, value in _linearize(name, value, n_structures, n_atoms).items():
-                if name in data["properties"]:
-                    # check that atom properties don't override user-provided
-                    # properties
-                    warnings.warn(
-                        f"ignoring the '{name}' atom property coming from the "
-                        "structures since it is already part of the properties"
-                    )
-                    continue
-
-                data["properties"][name] = value
-
-        for name, value in structure_properties(frames, composition).items():
-            _validate_property(name, value)
-            for name, value in _linearize(name, value, n_structures, n_atoms).items():
-                if name in data["properties"]:
-                    # check that frame properties don't override user-provided
-                    # properties or atom properties
-                    warnings.warn(
-                        f"ignoring the '{name}' structure property coming from the "
-                        "structures since it is already part of the properties"
-                    )
-                    continue
+        found_one_from_frame = False
+        atom_properties = _list_atom_properties(frames)
+        for name in atom_properties:
+            if name in data["properties"]:
+                found_one_from_frame = True
 
-                data["properties"][name] = value
+        structure_properties = _list_structure_properties(frames)
+        for name in structure_properties:
+            if name in data["properties"]:
+                found_one_from_frame = True
+
+        if not found_one_from_frame:
+            properties_list = ""
+
+            if len(structure_properties) != 0:
+                properties_list += "[" + ", ".join(structure_properties) + "]"
+
+            if len(atom_properties) != 0:
+                if len(properties_list) != 0:
+                    properties_list += " and "
+                properties_list += "[" + ", ".join(atom_properties) + "]"
+
+            if len(properties_list) != 0:
+                warnings.warn(
+                    "chemiscope behavior changed to no longer include properties "
+                    "from the structure objects. Use `chemiscope.extract_properties` "
+                    f"to also visualize these properties ({properties_list})"
+                )
```

Comment:
this warning will inform users of the change in behavior once they update to the new version. Hopefully it is clear enough!

Reply:
I am against having error messages that refer to previous behavior. Require keeping track to remove them at the next version bump, which I think we don't have the scale to do. I suggest that we don't show a message if properties are provided explicitly, and if someone only passes frames and asks for a map-containing visualization we could write something like "Structure or atom properties have to be provided using the `properties` keyword. If you are using an atoms frames that also contain properties, you can use `chemiscope.extract_properties` to retrieve them in a format suitable to be used in this context."

Reply:
Yeah I'm worried about too many warnings (that BTW don't disappear automatically in jupyter as far as I could tell). 
I don't have very strong feelings overall, I'd happily wait for you to converge on something that makes sense to you and then test it to give feedback.